CONTRIBUTING.md

```diff
@@ -3,9 +3,200 @@
+
+The *main* branch is where development happens. All the latest and
+greatest code, including breaking changes, happens on main. Currently patch releases are done from this branch
+only.
+
+The *release-X* branches will contain stable, backwards compatible code. On every
+major or minor release, a new branch will be created. It is from these
+branches that minor and patch releases will be tagged. In some cases, it may
+be necessary to open PRs for bugfixes directly against stable branches, but
+this should generally not be the case.
+
+## Contributing a Patch
+
+1. If you haven't already done so, sign a Contributor License Agreement (see
+   details above).
+1. Fork the desired repo, develop and test your code changes.
+1. Submit a pull request.
+
+All code PR must be labeled with one of
+
+- ⚠️ (`:warning:`, major or breaking changes)
+- ✨ (`:sparkles:`, feature additions)
+- 🐛 (`:bug:`, patch and bugfixes)
+- 📖 (`:book:`, documentation or proposals)
+- 🌱 (`:seedling:`, minor or other)
+
+Individual commits should not be tagged separately, but will generally be
+assumed to match the PR. For instance, if you have a bugfix in with
+a breaking change, it's generally encouraged to submit the bugfix
+separately, but if you must put them in one PR, mark the commit
+separately.
+
+All changes must be code reviewed. Coding conventions and standards are
+explained in the official [developer
+docs](https://github.com/kubernetes/community/tree/master/contributors/devel).
+Expect reviewers to request that you
+avoid common [go style
+mistakes](https://github.com/golang/go/wiki/CodeReviewComments) in your PRs.
+
+## Backporting a Patch
+
+Baremetal Operator does not yet maintain branches other than `main`. However, in future there will be
+release branches as well.
+
+For older versions through `release-X.Y` branches. We accept backports of bug fixes to the most recent
+release branch. For example, if the most recent branch is `release-0.2`, and the
+`main` branch is under active
+development for v0.3.0, a bug fix that merged to `main` that also affects
+`v0.2.x` may be considered for backporting
+to `release-0.2`. We generally do not accept PRs against older release branches.
+
+## Breaking Changes
+
+Breaking changes are generally allowed in the `main` branch.
+
+There may be times, however, when `main` is closed for breaking changes. This
+is likely to happen as we are close to release a new minor version.
+
+Breaking changes are not allowed in release branches, as these represent minor
+versions that have already been released.
+These versions have consumers who expect the APIs, behaviors, etc. to remain
+stable during the life time of the patch stream for the minor release.
+
+Examples of breaking changes include:
+
+- Removing or renaming a field in a CRD
+- Removing or renaming a CRD
+- Removing or renaming an exported constant, variable, type, or function
+- Updating the version of critical libraries such as controller-runtime,
```

Comment:
I assume, except for moving to a patch version?

Reply:
I didn't understand what you mean

Reply:
0.X.Y -> 0.X.(Y+1)?

Reply:
Yes, I will add it.